pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -114,7 +125,10 @@ def log_metrics(self, metrics: _OUT_DICT, step: Optional[int] = None) -> None:
             step = self.trainer.global_step
 
         # log actual metrics
-        self.trainer.logger.agg_and_log_metrics(scalar_metrics, step=step)
+        if self._override_agg_and_log_metrics:
+            self.trainer.logger.agg_and_log_metrics(metrics=scalar_metrics, step=step)
```

Comment:
for a separate PR: shouldn't we be calling `log_metrics` and `save` on *each* logger here?

Reply:
That's correct. Right now this works because `self.logger` is a `LoggerCollection` but this needs to be changed before `LoggerCollection` is removed.